pulp_rpm/app/tasks/publishing.py

```diff
@@ -257,6 +268,11 @@ def create_repomd_xml(content, publication, extra_repomdrecords,
     # Process all packages
     for package in packages.iterator():
         pkg = package.to_createrepo_c()
+        pkg_filename = os.path.basename(package.location_href)
+        pkg.location_href = os.path.join(
+            "Packages", pkg_filename[0].lower(),
+            pkg_filename
```

Comment:
So we are publishing using a filename and sorting it into directories bu their first letter.
With the recent suggestion to allow in one repo version packages with unique location_href, we might run into issues.
If one package is in `a/foo.rpm` and another is in `b/foo.rpm`, they will be allowed in a repo version because their location_hrefs differ but at publish time the original dir structure is ignored, so I guess only one of those files will be published. Any suggestions?
Should we check not the whole location_href but the filename during repo version creation?

 @ppicka, @ipanova , @dralley , @daviddavis , @dkliban

Reply:
i think we should use the pkg.location_href when publishing

Reply:
@goosemania i did some experiment and added location_href to the repo_key fields. We still cannot have 2 packages with a/foo.rpm and another  b/foo.rpm location_href because the check for `validate_version_paths` fails.  It checks if 2 CA have overlaped relative_path, which in our case does overlap because we do not take full location_href but just `os.path.basename`